app/master/cluster_master.py

```diff
@@ -55,6 +62,22 @@ def __init__(self):
 
         SlavesCollector.register_slaves_metrics_collector(lambda: self.all_slaves_by_id().values())
 
+    def configure_heartbeat(self):
+        self._heartbeat_job = self._scheduler.add_job(self.heartbeat, 'interval', seconds=self._heartbeat_frequency)
+        return self._scheduler
+
+    def heartbeat(self):
+        t = datetime.datetime.now()
+        slaves_to_disconnect = []
+        for slave in self._all_slaves_by_url.values():
+            if slave.is_alive() and not slave.is_responsive(t, self._heartbeat_frequency):
```

Comment:
Maybe it makes more sense to expose `slave.last_heartbeat_time()` and do the comparison here. It feels a bit strange to ask a `Slave` instance if it's responsive while also telling it the entire context for which it will be considered responsive.

Reply:
ah. Interesting point. Thanks!